### PR TITLE
*: remove the transaction kv count limit

### PR DESCRIPTION
**Pull request by jackysp**
Signed-off-by: Shuaipeng Yu <jackysp@gmail.com>

### What problem does this PR solve? 
It is extremely easy to reach the transaction kv count limit for TiDB. Which make the user have to modify their applications.

### What is changed and how it works?
Remove the transaction kv count limit, though keep the transaction size limit.

### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)

Code changes

 - Has exported variable/fields change

### CODE REVIEW

File: executor/seqtest/seq_executor_test.go [hunk 1]
```diff
@@ -793,6 +793,8 @@ func (s *seqTestSuite) TestCartesianProduct(c *C) {
 	plannercore.AllowCartesianProduct.Store(true)
 }
 
+// the entry count is removed, so the batch insert/delete may not used anymore.
```
Comment (zhangjinpeng1987) [inline executor/seqtest/seq_executor_test.go]: If this test is not used any more, please remove it directly.

Comment (jackysp) [reply executor/seqtest/seq_executor_test.go]: This PR is still WIP, plz wait.

**Comment by jackysp** [comment]
> Any performance test?

Some tests are being done, but not specifically for performance. I think this pr will not improve performance. We will do some profiling. However, these two days have been suspended because of the talent plan.

**Comment by disksing** [comment]
Maybe use some kind of `row count limit`? It will be understandable than `kv count limit`.

**Comment by zz-jason** [comment]
@jackysp any update? do we need to continue processing this PR?

**Comment by jackysp** [comment]
Yes, it is still WIP. I'm doing some tests for it, but the results may not be suitable to report to Github.

**Comment by coocood** [comment]
LGTM

**Comment by jackysp** [comment]
The unit test failed with the same reason as https://github.com/pingcap/tidb/issues/11693.

**Pull request merged**